test/support.ts

```diff
@@ -20,6 +20,27 @@ export async function submit(
   return receipt;
 }
 
+/// Submit a transaction and expect it to fail. Throws an error if it succeeds.
+export async function submitFailing(
+  tx: Promise<Ethers.ContractTransaction>
+): Promise<void> {
+  let succeeded = false;
+
+  try {
+    const receipt = await (await tx).wait();
+
+    if (receipt.status == 1) {
+      succeeded = true;
+    }
+  } catch {
+    /* OK */
+  }
+
+  if (succeeded) {
+    expect.fail("Expected transaction to fail");
+  }
```

Comment:
BTW how about this? Transaction failure always throws anyway.
```suggestion
  try {
    await (await tx).wait();
  } catch {
    return;
  }
  expect.fail("Expected transaction to fail");
```

Reply:
Yeah, cleaner.